integration/createdeletebeforeactive_test.go

```diff
@@ -0,0 +1,83 @@
+// +build integration
+
+package integration_test
+
+import (
+	"fmt"
+
+	awseks "github.com/aws/aws-sdk-go/service/eks"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/weaveworks/eksctl/pkg/ctl/cmdutils"
+	"github.com/weaveworks/eksctl/pkg/testutils/aws"
+	. "github.com/weaveworks/eksctl/pkg/testutils/matchers"
+)
+
+var _ = Describe("(Integration) Create & Delete before Active", func() {
+	const (
+		initNG = "ng-0"
+		testNG = "ng-1"
+	)
+
+	Describe("when creating a cluster with 1 node", func() {
+		var clName string
+		Context("for deleting a creating cluster", func() {
+			It("should run eksctl and not wait for it to finish", func() {
+
+				fmt.Fprintf(GinkgoWriter, "Using kubeconfig: %s\n", kubeconfigPath)
+
+				if clName == "" {
+					clName = cmdutils.ClusterName("", "") + "-delb4active"
+				}
+
+				eksctlStart("create", "cluster",
+					"--verbose", "4",
+					"--name", clName,
+					"--tags", "alpha.eksctl.io/description=eksctl delete before active test",
+					"--nodegroup-name", initNG,
+					"--node-labels", "ng-name="+initNG,
+					"--node-type", "t2.medium",
+					"--nodes", "1",
+					"--region", region,
+					"--version", version,
+				)
+			})
+		})
+
+		Context("when deleting the (creating) cluster", func() {
+
+			It("should not return an error", func() {
+
+				eksctlSuccess("delete", "cluster",
+					"--verbose", "4",
+					"--name", clName,
+					"--region", region,
+					"--wait",
+				)
+			})
+
+			It("and should have deleted the EKS cluster and both CloudFormation stacks", func() {
+
+				awsSession := aws.NewSession(region)
+
+				Expect(awsSession).ToNot(HaveExistingCluster(clName, awseks.ClusterStatusActive, version))
+
+				Expect(awsSession).ToNot(HaveExistingStack(fmt.Sprintf("eksctl-%s-cluster", clName)))
+				Expect(awsSession).ToNot(HaveExistingStack(fmt.Sprintf("eksctl-%s-nodegroup-ng-%d", clName, 0)))
```

Comment:
shouldn't this technically be `fmt.Sprintf("eksctl-%s-nodegroup-%s", clName, initNG)`? are there methods we can re-use to build that string that eksctl uses itself?

Reply:
There is a method in pkg/cfn/manager/nodegroup.go:
```
func (c *StackCollection) makeNodeGroupStackName(name string) string {
	return fmt.Sprintf("eksctl-%s-nodegroup-%s", c.spec.Metadata.Name, name)
}
```
As you can see, it's not an exported outside of its package.  Essentially though it is 'eksctl-<clusterName>-nodegroup-<nodegroup name>, so in this case, the issue with the test is that we're generating "eksctl-<clustername>-nodegroup-ng-ng-0".  It was never going to match an existing nodegroup!

Reply:
I don't think so, the test builds the right string